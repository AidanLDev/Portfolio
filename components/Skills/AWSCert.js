import React from 'react'
import { Image, Tooltip } from '@chakra-ui/react'

export default function AWSCert() {
  return (
    <>
      <a href="https://www.credly.com/badges/aeaec428-0a39-4245-8bb6-67851fa02904/public_url" rel="noreferrer" target="_blank">
        <Tooltip label="Click to verify certificate">
          <Image src='/images/Certs/AWSCPBadge.png' alt="AWS Cloud Practitioner Badge" />
        </Tooltip>
      </a>
    </>
  )
}
