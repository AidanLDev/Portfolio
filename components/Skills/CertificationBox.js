import { Image } from '@chakra-ui/image'
import { Tooltip } from '@chakra-ui/react'
import React from 'react'
import { motion } from 'framer-motion'

import styles from './styles.module.scss'

export default function CertificationBox({ name, img, link }) {
  return (
    <div className={styles.certBox}>
      <a href={link} rel="noreferrer" target="_blank">
        <Tooltip label="Click to verify certificate">
          <Image
            as={motion.img}
            alt={name}
            src={`/images/Certs/${img}`}
            w={500}
            h={400}
            whileHover={{ scale: 1.05 }}
          />
        </Tooltip>
      </a>
    </div>
  )
}
