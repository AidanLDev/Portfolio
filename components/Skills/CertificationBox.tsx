import { Image } from '@chakra-ui/react'
import { Tooltip } from '@chakra-ui/react'
import React from 'react'
import { motion } from 'framer-motion'

import styles from './styles.module.scss'

export interface CertBoxProps {
  name: string
  img: string
  link: string
}

export default function CertificationBox({ name, img, link }: CertBoxProps) {
  return (
    <div className={styles.certBox}>
      <a href={link} rel="noreferrer" target="_blank">
        <Tooltip label="Click to verify certificate">
          <Image
            as={motion.img}
            alt={name}
            src={`/images/Certs/${img}`}
            w={500}
            h={400}
            whileHover={{ scale: 1.01 }}
          />
        </Tooltip>
      </a>
    </div>
  )
}
