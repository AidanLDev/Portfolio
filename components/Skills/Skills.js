import React from 'react'

import styles from './styles.module.scss'
import SkillBox from './SkillBox'

const skills = [
  {
    name: 'AWS',
    logo: '/images/Tech/AWSLogo.png',
  },
  {
    name: 'JavaScript',
    logo: '/images/Tech/JavaScriptLogo.png',
  },
  {
    name: 'TypeScript',
    logo: '/images/Tech/TypeScript.png',
  },
  {
    name: 'React',
    logo: '/images/Tech/ReactLogo.png',
  },
  {
    name: 'Node',
    logo: '/images/Tech/Node.png',
  },
  {
    name: 'MySQL',
    logo: '/images/Tech/mySQLLogo.png',
  },
  {
    name: 'SASS',
    logo: '/images/Tech/sassLogo.png',
  },
  {
    name: 'Tailwind',
    logo: '/images/Tech/Tailwind.png',
  },
  {
    name: 'Angular',
    logo: '/images/Tech/AngularLogo.png',
  },
  {
    name: 'Astro',
    logo: '/images/Tech/astro.png',
  },
  {
    name: 'Next',
    logo: '/images/Tech/Next.png',
  },
  {
    name: 'GIT',
    logo: '/images/Tech/GitLogo.png',
  },
]

export default function Skills() {
  return (
    <div className={styles.skillsList}>
      {/* AWS JS TS          */}
      {skills.map((skill, idx) => (
        <SkillBox
          key={`${idx}__${skill.name}`}
          logo={skill.logo}
          name={skill.name}
        />
      ))}
    </div>
  )
}
