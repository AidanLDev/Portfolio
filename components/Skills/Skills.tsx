import React from 'react'
import SkillBox from './SkillBox'
import { skills } from './skillsList'

import styles from './styles.module.scss'

export default function Skills() {
  return (
    <div className={styles.skillsList}>
      {skills.map((skill, idx) => (
        <SkillBox
          key={`${idx}__${skill.name}`}
          logo={skill.logo}
          name={skill.name}
        />
      ))}
    </div>
  )
}
