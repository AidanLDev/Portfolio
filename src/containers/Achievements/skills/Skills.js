import React from 'react'

import styles from './Style'

/*  Logos */
import JSLogo from 'assets/Tech/JavaScriptLogo.png'
import ReactLogo from 'assets/Tech/ReactLogo.png'
import AngularLogo from 'assets/Tech/AngularLogo.png'
import SASSLogo from 'assets/Tech/sassLogo.png'
import NodeLogo from 'assets/Tech/Node.png'
import MySQLLogo from 'assets/Tech/mySQLLogo.png'
import AWSLogo from 'assets/Tech/AWSLogo.png'

const Skills = props => {

  const imgHeight = "50";
  const imgWidth = "50";

  return (
    <div>
      <h1>Skills</h1>
        <div className={styles.skillsContainer}>
          <div>
            <h2>JavaScript <img src={JSLogo} height={imgHeight} width={imgWidth} alt="JavaScript Logo"/></h2>
            <p>From front-end frameworks to using JQuery to vanilla JavaScript when writting my first web applications. JavaScript has been a constant in my web development carear.</p>
          </div>
          <div>
            <h2>React <img src={ReactLogo} height={imgHeight} width={imgWidth} alt="React Logo"/></h2>
            <p>The front-end framework I use within my organisation to create CMS, Callback Solutions and web applications</p>
          </div>
          <div>
            <h2>Angular <img src={AngularLogo} height={imgHeight} width={imgWidth} alt="Angular Logo"/></h2>
            <p>Developed public facing forms for finacial(check spelling) services using this framework. Creating a front-end that looks good both on 300px wide mobiles and 1920px desktops</p>
          </div>
          <div>
            <h2>SASS <img src={SASSLogo} height={imgHeight} width={imgWidth} alt="SASS Logo"/></h2>
            <p>CSS prefixing. Re-usable colour pallets/variables/mixins simplifies consistancy</p>
          </div>
          <div>
            <h2>Node.js <img src={NodeLogo} height={imgHeight} width={imgWidth} alt="Node Logo"/></h2>
            <p>Used in conjunction with Koa and Express to server secure services.</p>
          </div>
          <div>
            <h2>MySQL <img src={MySQLLogo} height={imgHeight} width={imgWidth} alt="MySQL Logo"/></h2>
            <p>The RDMS used for persitant data storage, experiance in both Oracle database engine and MS SQL creating transactional SQL procedures.</p>
          </div>
          <div>
            <h2>AWS <img src={AWSLogo} height={imgHeight} width={imgWidth} alt="AWS Logo"/></h2>
            <p>The cloud platform used to host this portfolio and my organisations mission criticle applications</p>
          </div>
        </div>
    </div>
  );
}

export default Skills;