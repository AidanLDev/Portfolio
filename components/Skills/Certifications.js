import React from 'react'

import CertificationBox from './CertificationBox'

import style from './styles.module.scss'

const certs = [{ name: 'Udemy React 16 Course', img: 'udemyBurgerBuilder.jpg', link: 'https://www.udemy.com/certificate/UC-A5AOU8C4/' }, { name: 'AWS Cloud Practitioner', img: 'AWSCPBadge.png', link: 'https://www.credly.com/badges/aeaec428-0a39-4245-8bb6-67851fa02904/public_url ' }, { name: 'Free Code Camp JS course', img: 'FreeCodeCampCert.png', link: 'https://www.freecodecamp.org/certification/fcc99c3375c-a5d4-4e8a-ba4a-96473ca87aa8/javascript-algorithms-and-data-structures' }]

export default function Certifications() {
  return (
    <div className={style.certContainer}>
      {certs.map((cert, idx) => <CertificationBox key={`${cert.name}__${idx}`} name={cert.name} img={cert.img} link={cert.link} />)}
    </div>
  )
}
