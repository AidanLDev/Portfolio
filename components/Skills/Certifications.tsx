import React from 'react'
import CertificationBox from './CertificationBox'
import { certs } from './CertificationsList'

import style from './styles.module.scss'

export default function Certifications() {
  return (
    <div className={style.certContainer}>
      {certs.map((cert, idx) => (
        <CertificationBox
          key={`${cert.name}__${idx}`}
          name={cert.name}
          img={cert.img}
          link={cert.link}
        />
      ))}
    </div>
  )
}
