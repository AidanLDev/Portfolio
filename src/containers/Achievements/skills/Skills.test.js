import React from 'react';
import Skills from './Skills';
import { shallow } from 'enzyme';

it('Renders Skills Component Correctly', () => {
  const component = shallow(<Skills />);
  expect(component).toMatchSnapshot();
});
