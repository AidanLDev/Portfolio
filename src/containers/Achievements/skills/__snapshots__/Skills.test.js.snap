// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Skills Component Correctly 1`] = `
<div>
  <h1>
    Technology Stack
  </h1>
  <div
    className="skillsContainer"
  >
    <div>
      <img
        alt="JavaScript Logo"
        height="50"
        src="JavaScriptLogo.png"
        width="50"
      />
      <h2>
        JavaScript
      </h2>
    </div>
    <div>
      <img
        alt="TypeScript Logo"
        height="50"
        src="TypeScript.png"
        width="50"
      />
      <h2>
        TypeScript
      </h2>
    </div>
    <div>
      <img
        alt="React Logo"
        height="50"
        src="ReactLogo.png"
        width="50"
      />
      <h2>
        React
      </h2>
    </div>
    <div>
      <img
        alt="Angular Logo"
        height="50"
        src="AngularLogo.png"
        width="50"
      />
      <h2>
        Angular
      </h2>
    </div>
    <div>
      <img
        alt="SASS Logo"
        height="50"
        src="sassLogo.png"
        width="50"
      />
      <h2>
        SASS
      </h2>
    </div>
    <div>
      <img
        alt="Node Logo"
        height="50"
        src="Node.png"
        width="50"
      />
      <h2>
        Node.js
      </h2>
    </div>
    <div>
      <img
        alt="MySQL Logo"
        height="50"
        src="mySQLLogo.png"
        width="50"
      />
      <h2>
        MySQL
      </h2>
    </div>
    <div>
      <img
        alt="MongoDB Logo"
        height="50"
        src="mongoDBLogo.png"
        width="50"
      />
      <h2>
        MongoDB
      </h2>
    </div>
    <div>
      <img
        alt="AWS Logo"
        height="50"
        src="AWSLogo.png"
        width="50"
      />
      <h2>
        AWS
      </h2>
    </div>
    <div>
      <img
        alt="Git Logo"
        height="50"
        src="GitLogo.png"
        width="50"
      />
      <h2>
        GIT
      </h2>
    </div>
    <div>
      <img
        alt="Webpack Logo"
        height="50"
        src="webpackLogo.png"
        width="50"
      />
      <h2>
        Webpack
      </h2>
    </div>
    <div>
      <img
        alt="Docker Logo"
        height="50"
        src="DockerLogo.png"
        width="50"
      />
      <h2>
        Docker
      </h2>
    </div>
  </div>
</div>
`;
